routes/index.js

```diff
@@ -1,279 +1,43 @@
-        res.write(chunk)
-        res.write('\r\n--cctv\r\n')
-        callback()
-      }
-    })
-
-    res.status(200)
-      .set('Content-Type', 'multipart/x-mixed-replace;boundary=cctv')
-      .write('--cctv\r\n')
-    writable.write(jpeg, { end: true })
-    pipe2jpeg.pipe(writable)
-
-    res.once('close', () => {
-      if (pipe2jpeg && writable) {
-        pipe2jpeg.unpipe(writable)
-      }
-      res.end()
-    })
-  })
-
-  app.get(`/api/*/mp4/video.mp4`, (req, res) => {
-    // console.log('video mp4')
-    const mp4frag = res.locals.mp4frag
-    const init = mp4frag.initialization
-    if (!init) {
-      return res.status(503)
-        .set('Retry-After', 10)
-        .end(`Stream "${res.locals.id}" video.mp4 currently unavailable. Please try again later.`)
-    }
-
-    res.status(200)
-      .set('Content-Type', 'video/mp4')
-      .write(init)
-    const segment = mp4frag.segment
-    if (segment) {
-      res.write(segment)
-    }
-    mp4frag.pipe(res, { end: true })
-    res.once('close', () => {
-      if (mp4frag) {
-        mp4frag.unpipe(res)
-      }
-      res.end()
-    })
-  })
-
-  app.get(`/api/*/mp4/playlist.m3u8`, (req, res) => {
-    const mp4frag = res.locals.mp4frag
-    const m3u8 = mp4frag.m3u8
-    if (!m3u8) {
-      return res.status(503)
-        .set('Retry-After', 10)
-        .end(`Stream "${res.locals.id}" playlist.m3u8 currently unavailable. Please try again later.`)
-    }
-    res.status(200)
-      .set('Content-Type', 'application/vnd.apple.mpegurl')
-      .end(m3u8)
-  })
-
-  app.get(`/api/*/mp4/init-[a-zA-Z]+.mp4`, (req, res) => {
-    const mp4frag = res.locals.mp4frag
-    const init = mp4frag.initialization
-    if (!init) {
-      return res.status(503)
-        .set('Retry-After', 10)
-        .end(`Stream "${res.locals.id}" init mp4 currently unavailable. Please try again later.`)
-    }
-    res.status(200)
-      .set('Content-Type', 'video/mp4')
-      .end(init)
-  })
-
-  app.get(`/api/*/mp4/:segment(*.m4s)`, (req, res) => {
-    // console.log('named segment', req.params.segment)
-    const mp4frag = res.locals.mp4frag
-    const segment = mp4frag.getHlsNamedSegment(req.params.segment)
-    if (!segment) {
-      return res.status(503)
-        .end('503 for m4s segment')
-    }
-    res.status(200)
-      .set('Content-Type', 'video/mp4')
-      .end(segment)
-  })
+  app.use('/api/*/jpeg', require('./jpeg/router'))
+  app.use('/api/*/mp4', require('./mp4/router'))
 
-  app.get(`/api/*/mp4/playlist.m3u8.txt`, (req, res) => {
-    const mp4frag = res.locals.mp4frag
-    const m3u8 = mp4frag.m3u8
-    if (!m3u8) {
-      return res.status(503)
-        .set('Retry-After', 10)
-        .end(`Stream "${res.locals.id}" playlist.m3u8 currently unavailable. Please try again later.`)
-    }
-    res.status(200)
-      .set('Content-Type', 'text/plain')
-      .end(m3u8)
-  })
+  app.get('/api/*/cmd/:action', (req, res) => {
```

Comment:
This whole thing looked kinda weird, it seemed to ignore the `ffmpeg.running` flag. So I updated so it would run the command if ffmpeg was actually running to stop and vice versa for not running to start. Not sure if there was a reason for this.